extensions/dicom-microscopy/src/tools/viewerManager.js

```diff
@@ -0,0 +1,450 @@
+import Publisher from '../utils/Publisher';
+import coordinateFormatScoord3d2Geometry from '../utils/coordinateFormatScoord3d2Geometry';
+import styles from '../utils/styles';
+
+// Events from the third-party viewer
+const ApiEvents = {
+  ROI_ADDED: 'dicommicroscopyviewer_roi_added',
+  ROI_MODIFIED: 'dicommicroscopyviewer_roi_modified',
+  ROI_REMOVED: 'dicommicroscopyviewer_roi_removed',
+};
+
+const EVENTS = {
+  ADDED: 'added',
+  MODIFIED: 'modified',
+  REMOVED: 'removed',
+  UPDATED: 'updated',
+};
+
+/**
+ * ViewerManager encapsulates the complexity of the third-party viewer and
+ * expose only the features/behaviors that are relevant to the application
+ */
+class ViewerManager extends Publisher {
```

Comment:
same for this, it makes sense to have them registered in services manager of ohif instead of re-deising the architecture